### ISSUE TITLE
Multiple choice grading (QCM) answer options  with the annotated template are not consistently applied during grading

### ISSUE THREAD

**Issue by sderrien** [correctexam/corrigeExamFront#459]
The annotated question template

<img width="1118" alt="image" src="https://github.com/correctexam/corrigeExamFront/assets/1744645/3c00e54c-0400-415c-8e24-c65576f9126e">

The grading choice corresponding to Q3 shows options that are not present in the template (.

<img width="1322" alt="image" src="https://github.com/correctexam/corrigeExamFront/assets/1744645/340cb4c3-613a-4462-a601-34de0662f437">

More broadly, is it expected that changes in the answer regex be immediately applied to the grading at hand  ?

**Comment by barais** [comment]
The best for QCM is to select for a first try just the place where you get the square. 

You can enlarge it later before sending result to student.

**Comment by sderrien** [comment]
My description was probably not very clear. The problem is that I observe choices (a & e) for grading for which I have no idea where they come from (on the template, I only allow one valid expression)

**Comment by barais** [comment]
Yes, your description was clear. But it means that in analysing the template, the qcm algo found 5 square (that is why it could be a good idea to reduce the search space for square in creating a smaller zone, you could increase it later). 

The qcm algo is simple:
1: search square in template from question zone.
2: search empty square in scan for the zone where it detect square in template
3: perform a new alignment based on the match of void square. 
4: compare the density of the square zones in the scan for each copy. 
5: send the answer. 

That is why, if it creates a 'e' comment. It means that it detect 5 squares

Regards

Olivier

**Comment by sderrien** [comment]
OK, now I get it. 

Then I have two suggestions :
- (optionnally) ask the user to specify the number of distinct check-boxes to allow early detection of inconsistencies in the template.
- an option during the grading process to "enlarge" by 20% the answer area to accommodate with poorly aligned scans (I agree one can switch to the whole page but this can be cumbersome)

Regards,

Steven
